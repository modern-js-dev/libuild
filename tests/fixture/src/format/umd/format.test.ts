import { getLibuilderTest } from '@modern-js/libuild-test-toolkit';

describe('fixture:format:umd', () => {
  it('format umd', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      format: 'umd',
      splitting: true,
    });
    await bundler.build();

    bundler.expectJSOutputMatchSnapshot();
  });
});
