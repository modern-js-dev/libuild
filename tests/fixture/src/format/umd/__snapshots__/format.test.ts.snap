// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixture:format:umd format umd 1`] = `
"(function (global, factory) {
  if (typeof define === \\"function\\" && define.amd) {
    define([\\"exports\\"], factory);
  } else if (typeof exports !== \\"undefined\\") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.main = mod.exports;
  }
})(typeof globalThis !== \\"undefined\\" ? globalThis : typeof self !== \\"undefined\\" ? self : this, function (_exports) {
  \\"use strict\\";

  Object.defineProperty(_exports, \\"__esModule\\", {
    value: true
  });
  _exports.a = a;

  // index.ts
  function a() {
    console.log(1);
  }
});"
`;
